File: 1122Programming/leetcode/231. Power of Two/ans.cpp
/*
 *
 * https://leetcode.com/problems/power-of-two/
 * Problem ID: 231. Power of Two
 * 
 * Author: Ateto
 * 
 */

class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n <= 0) return false;
        while(n != 1) {
            if(n & 1) return false;
            n >>= 1;
        }
        return true;
    }
};

File: 1122Programming/leetcode/231. Power of Two/ans.cs
/*
 *
 * https://leetcode.com/problems/power-of-two/
 * Problem ID: 231. Power of Two
 * 
 * Author: asd7766zxc
 * 
 */

public class Solution {
    public bool IsPowerOfTwo(int n) {
        return Convert.ToString(n, 2).Select(i => i -'0').Sum() == 1 && n > 0;
    }
}